### DISCUSSION TITLE
Is there support for multi-package repos?

### DISCUSSION THREAD

**Discussion by TheYarin** [fwouts/previewjs#63] [Q&A]
I am trying to use this amazing extension with a repo that contains multiple sub-packages - subfolders with their own `package.json` file, `node_modules`, etc. Is there a way to define a `reactpreview.config.js` in the repo's root to apply for all the packages? When I tried simply doing that, the aliases didn't work properly. I suspect the project's root is assumed to be where the first `package.json` is found up the folder hierarchy.

**Comment by fwouts** [comment]
Hi @TheYarin! You're correct. Just to confirm, until that's possible, I imagine you could create multiple reactpreview.config.js files but it would become quite cumbersome?

**Reply by TheYarin** [reply]
Yep, giant project, dozens of packages - quite cumbersome.

**Reply by fwouts** [reply]
Appreciate the insight. I'm a big fan of monorepos myself, so I see the appeal in having this feature :) Added to the todo list!

**Reply by TheYarin** [reply]
Thank you! Having this extension in a large project would be super useful.

**Reply by TheYarin** [reply]
I suspect React Preview is using the current package's `node_modules` (thus missing out on everything on other node_modules folders in the project) and that's what causing my preview attempts to fail. Is this the case?

**Reply by fwouts** [reply]
I'm not entirely sure! Under the hood, node modules resolution is handled by Vite. Basically, if you can find a way to get Vite to work for one of these subpackages, we should be able to make React Preview work for it too.

I'm just about to release Preview.js, the cross-framework and open source version of React Preview, within the next week or so, so you'll be able to see exactly how it works. Adding monorepo support should be soon on the list after that.
